src/gebt_poc/gen_alpha_2D.cpp

```diff
@@ -0,0 +1,334 @@
+        lagrange_mults_next
+    );
+
+    if (this->is_converged_) {
+        log->Info(
+            "Newton-Raphson iterations converged in " + std::to_string(n_iterations + 1) +
+            " iterations\n"
+        );
+        return results;
+    }
+
+    log->Warning(
+        "Newton-Raphson iterations failed to converge on a solution after " +
+        std::to_string(n_iterations + 1) + " iterations!\n"
+    );
+
+    return results;
+}
+
+void GeneralizedAlphaTimeIntegrator::UpdateGeneralizedCoordinates(
+    Kokkos::View<const double* [kNumberOfLieGroupComponents]> gen_coords,
+    Kokkos::View<const double* [kNumberOfLieAlgebraComponents]> delta_gen_coords,
+    Kokkos::View<double* [kNumberOfLieGroupComponents]> gen_coords_next
+) {
+    const auto h = this->time_stepper_.GetTimeStep();
+
+    // Loop over all nodes in the system and update the generalized coordinates
+    for (size_t node = 0; node < gen_coords.extent(0); node++) {
+        auto update_generalized_coordinates = KOKKOS_LAMBDA(size_t) {
+            // {gen_coords_next} = {gen_coords} + h * {delta_gen_coords}
+            //
+            // Step 1: R^3 update, done with vector addition
+            auto current_position = gen_alpha_solver::Vector{
+                gen_coords(node, 0), gen_coords(node, 1), gen_coords(node, 2)};
+            auto updated_position = gen_alpha_solver::Vector{
+                delta_gen_coords(node, 0), delta_gen_coords(node, 1), delta_gen_coords(node, 2)};
+            auto r = current_position + (updated_position * h);
+
+            // Step 2: SO(3) update, done with quaternion composition
+            gen_alpha_solver::Quaternion current_orientation{
+                gen_coords(node, 3), gen_coords(node, 4), gen_coords(node, 5), gen_coords(node, 6)};
+            auto updated_orientation = gen_alpha_solver::quaternion_from_rotation_vector(
+                // Convert Vector -> Quaternion via exponential mapping
+                gen_alpha_solver::Vector{
+                    delta_gen_coords(node, 3), delta_gen_coords(node, 4),
+                    delta_gen_coords(node, 5)} *
+                h
+            );
+            auto q = current_orientation * updated_orientation;
+
+            gen_coords_next(node, 0) = r.GetXComponent();
+            gen_coords_next(node, 1) = r.GetYComponent();
+            gen_coords_next(node, 2) = r.GetZComponent();
+            gen_coords_next(node, 3) = q.GetScalarComponent();
+            gen_coords_next(node, 4) = q.GetXComponent();
+            gen_coords_next(node, 5) = q.GetYComponent();
+            gen_coords_next(node, 6) = q.GetZComponent();
+        };
+        Kokkos::parallel_for(1, update_generalized_coordinates);
+    }
+}
+
+bool GeneralizedAlphaTimeIntegrator::CheckConvergence(const Kokkos::View<double*> residual) {
+    // L2 norm of the residual vector should be very small (< epsilon) for the solution
```

Comment:
The L2 norm of the residual can be calculated as `KokkosBlas::nrm2(residual)` just to save some code. May consider eliminating the function since it's really easy to check.

Reply:
I actually like the separate function, to improve readability.  I was planning to refactor this function at some point to
```
return KokkosBlas::nrm2(residual) < kConvergenceTolerance;
```